exasol/nb_connector/text_ai_extension_wrapper.py

```diff
@@ -58,59 +83,63 @@
-        project_wheel, slc_tar_gz = unzipped_files
-        ...
+class ModelInstaller:
+    """
+    Download Huggingface models and deploy them to Exasol BucketFS.
     """
 
-    zip_url = conf.get(CKey.text_ai_pre_release_url)
-    if not zip_url:
-        raise ValueError("Pre-release URL is not set.")
-    zip_password = conf.get(CKey.text_ai_zip_password)
-    if not zip_password:
-        raise ValueError("Pre-release zip password is not set.")
-
-    # Download the file
-    response = requests.get(zip_url, stream=True)
-    response.raise_for_status()
-
-    with tempfile.NamedTemporaryFile() as tmp_file:
-        # Save the downloaded zip in a temporary file
-        for chunk in response.iter_content(chunk_size=1048576):
-            tmp_file.write(chunk)
-        tmp_file.flush()
-        with tempfile.TemporaryDirectory() as tmp_dir:
-            # Unzip the file into a temporary directory
-            unzip_cmd = [
-                "unzip",
-                "-q",
-                "-P",
-                zip_password,
-                tmp_file.name,
-                "-d",
-                tmp_dir,
-            ]
-            subprocess.run(unzip_cmd, check=True, capture_output=True)
-            tmp_path = Path(tmp_dir)
-            # Find and return the project wheel and the SLC
-            project_wheel = next(tmp_path.glob("*.whl"))
-            slc_tar_gz = next(tmp_path.glob("*.tar.gz"))
-            conf.save(CKey.txaie_slc_file_local_path, str(slc_tar_gz))
-            yield project_wheel, slc_tar_gz
-
-
-def deploy_licence(
+    def __init__(self, bucketfs_location: bfs.path.PathLike, bfs_subdir: str):
+        self.bucketfs_location = bucketfs_location
+        self.bfs_subdir = bfs_subdir
+        self.models = [
+            TransformerModel(
+                DEFAULT_FEATURE_EXTRACTION_MODEL,
+                "feature-extraction",
+                AutoModel,
+            ),
+            TransformerModel(
+                DEFAULT_NAMED_ENTITY_MODEL,
+                "token-classification",
+                AutoModelForTokenClassification,
+            ),
+            TransformerModel(
+                DEFAULT_NLI_MODEL,
+                "zero-shot-classification",
+                AutoModelForSequenceClassification,
+            ),
+        ]
+
+    def download_and_install(self) -> None:
+        for model in self.models:
+            with yaspin(text=f"- Huggingface model {model}") as spinner:
+                download_transformers_model(
+                    bucketfs_location=self.bucketfs_location,
+                    sub_dir=self.bfs_subdir,
+                    task_type=model.task_type,
+                    model_name=model.name,
+                    model_factory=model.factory,
+                )
+            spinner.ok(CHECKMARK)
+
+
+def deploy_license(
     conf: Secrets,
-    licence_file: Optional[Path] = None,
-    licence_content: Optional[str] = None,
+    # license_file: Optional[Path] = None,
```

Comment:
As getting the text from a path is so simple, I would remove the support for passing a file, here. 

Additionally: We could think of a GUI in a Jupyter notebook, allowing the user to upload a license file.
In this case also only the text content of the license file needs to be processed, IMHO.